### PR TITLE
fixes for reinstall, docs, error catching

### LINKED ISSUE #206: Array size checking fails on (n,1)-shaped arrays (py3)
Hi Erin,

Looks like you've made an impressive amount of progress toward 1.0 -- congrats!

This turned up last night (my CI build suddenly started failing!).

It looks like arrays of size:
 (N,) works
 (N,N) works
 (1,N) works
 (N,1) fails

This last case occurs in the wild (in my code, eg) because we have a "fluxes" column that includes measured fluxes for all available bands, and sometimes there is only one band available.

Test case:
```
import fitsio
import numpy as np
print('numpy', np.__version__)
fitsio.write('x.fits', names=['a'], data=[np.zeros((2,1))])
```

Result (with some commented-out print statements in that vicinity enabled):
```
> python testcase.py
numpy 1.14.5
col_tdim [1]
column name: a
(2, 1)
column dtype: f8
input dtype: f8
column shape: None
input shape: (1,)

Traceback (most recent call last):
  File "testcase.py", line 4, in <module>
    fitsio.write('x.fits', names=['a'], data=[np.zeros((2,1))])
  File "/usr/local/lib/python3.7/site-packages/fitsio/fitslib.py", line 263, in write
    **keys)
  File "/usr/local/lib/python3.7/site-packages/fitsio/fitslib.py", line 509, in write
    write_bitcols=write_bitcols)
  File "/usr/local/lib/python3.7/site-packages/fitsio/fitslib.py", line 818, in write_table
    self[-1].write(data,names=names)
  File "/usr/local/lib/python3.7/site-packages/fitsio/fitslib.py", line 1693, in write
    self._verify_column_data(tcolnum,tdata)
  File "/usr/local/lib/python3.7/site-packages/fitsio/fitslib.py", line 1815, in _verify_column_data
    "expected '%s', got '%s'" % (col_name,col_shape, this_shape))
ValueError: bad input shape for column 'a': expected 'None', got '(1,)'
```
```
> python --version
Python 3.7.2
```

OHH, and interestingly, it seems to work in python2:
```
> python2 testcase.py
('numpy', '1.14.0')

> python2 --version
Python 2.7.15
```

### PR DESCRIPTION
**Pull request by beckermr**
This PR fixes

- repatching is now tried and exceptions are caught
- shape testing for columns catches a corner case (closes #206)
- adds easier testing docs

### CODE REVIEW

**Comment by beckermr** [comment]
@dstndstn @esheldon I have put a fixe for #206 here. Can y'all take it for a test drive? The tests pass for me locally.

**Comment by beckermr** [comment]
@esheldon this also fixes the repatching issues you were having in the install

**Comment by esheldon** [comment]
Thanks  Matt.  Can you please point me to the substantial changes?    Its hard to wade through all the formatting changes.

**Comment by beckermr** [comment]
Sorry. You can actually turn off white space changes in the diff settings on GitHub. I’ll leave a note.

File: fitsio/fitslib.py [hunk 1]
```diff
@@ -1760,7 +1760,9 @@ def _verify_column_data(self, colnum, data):
         """
         this_dt = data.dtype.descr[0]
 
-        if len(data.shape) > 1:
+        if len(data.shape) > 2:
+            this_shape = data.shape[1:]
+        elif len(data.shape) == 2 and data.shape[1] > 1:
```
Comment (beckermr) [inline fitsio/fitslib.py]: This is the main one @esheldon

File: setup.py [hunk 2]
```diff
@@ -151,9 +151,18 @@ def patch_cfitsio(self):
...
+                    'patch -N --dry-run %s/%s %s' % (
+                        self.cfitsio_build_dir, fname, patch),
+                    shell=True)
+            except subprocess.CalledProcessError:
+                pass
+            else:
+                subprocess.check_call(
+                    'patch %s/%s %s' % (
+                        self.cfitsio_build_dir, fname, patch),
+                    shell=True)
```
Comment (beckermr) [inline setup.py]: This block fixes the patching.

**Comment by esheldon** [comment]
Thanks, I figure out how to hide whitespace changes

File: fitsio/fitslib.py (hunk 1 above)
Comment (esheldon) [reply fitsio/fitslib.py]: I understand this now.  I thought we were dealing with a different issue.

Comment (esheldon) [reply fitsio/fitslib.py]: this makes sense I think

File: setup.py (hunk 2 above)
Comment (esheldon) [reply setup.py]: I'll have to take your word for it

Comment (beckermr) [reply setup.py]: I tested a bunch of different ways of reinstalling in different orders. It appears to work on my mac at least.
